compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -11489,6 +11514,16 @@ private STNode parseDoubleRightArrow() {
         }
     }
 
+    private STNode parseDoubleRightArrow(ParserRuleContext parserRuleContext) {
+        STToken token = peek();
+        if (token.kind == SyntaxKind.RIGHT_DOUBLE_ARROW_TOKEN) {
+            return consume();
+        } else {
+            recover(token, parserRuleContext);
+            return parseDoubleRightArrow();
```

Comment:
```suggestion
            return parseDoubleRightArrow(parserRuleContext);
```

Also, Let's get rid of the other `parseDoubleRightArrow()` method.